app/presenters/document_presenter.rb

```diff
@@ -8,13 +8,31 @@ class DocumentPresenter
     :bulk_published,
     to: :"document.details"
 
+  attr_reader :finder
+
   def initialize(finder, document)
     @finder = finder
     @document = document
   end
 
+  def respond_to?(method)
+   if finder.facets.map(&:key).include?(method.to_s)
+     true
+   else
+     super
+   end
+ end
+
+  def method_missing(method, *args)
+    if finder.facets.map(&:key).include?(method.to_s)
+      document.details.try(method)
```

Comment:
Ah, yep. Use [`Object#send`](http://ruby-doc.org/core-2.0.0/Object.html#method-i-send) to dynamically call a method. [`Object#try`](http://guides.rubyonrails.org/active_support_core_extensions.html#try) is a Rails extension to make avoid having to do `nil` checking.